### ISSUE TITLE
ipv6 selection

### ISSUE THREAD

**Issue by Bryannn** [rg3/youtube-dl#9601]
I see there is an option --force-ipv6 to use ipv6, does it choose a random ipv6 ip from my range ?

Thanks!

**Comment by yan12125** [comment]
It chooses "the best match" IPv6 address on your device, which depends on your routing table.

**Comment by Bryannn** [comment]
I asked this because my ipv4 got blocked by YouTube for sending too many requests. So now I don't want it to use just 1 IP for downloading. If it chose a random IP out of the 256 IP's, would have been great. what do you exactly mean by best match. Could the best match keep changing each time it downloads? Thanks!

**Comment by Bryannn** [comment]
is there a way to know which ip is using ? cant see the used ip in the verbose mode

**Comment by yan12125** [comment]
>  what do you exactly mean by best match.

See https://en.wikipedia.org/wiki/Routing_table.

> Could the best match keep changing each time it downloads?

In general no.

**Comment by yan12125** [comment]
> is there a way to know which ip is using ?

Add ```--call-home```

And by the way, there are several "too many requests" reports and others' solution may help: #5138 and [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#http-error-429-too-many-requests-or-402-payment-required)

**Comment by Bryannn** [comment]
I really appreciate all your help. I've already read that post. I just have one last small question. So what if the best match IP gets blocked by YouTube for sending too many requests. Will it move to another IP from the range or just stop there? Thanks :)

**Comment by yan12125** [comment]
It will just get blocked. youtube-dl is not intelligent enough to know the next unblocked IP :)

**Comment by Bryannn** [comment]
Can someone here help me change the code of youtube-dl so that it chooses a random IP each time from my ipv6 range instead of just using the best one. I could pay for it.

**Comment by yan12125** [comment]
FYI: you can use ```--source-address``` to force the IP to use. You may need to write a script to choose from available IPs and assign it to youtube-dl.

**Comment by Bryannn** [comment]
Yes that's what i'm talking about. If some here could help me do that.

**Comment by yan12125** [comment]
Here's an issue: usually YouTube blocks a subnet (an IP range) rather than a single IP, so random selection may not be useful.

**Comment by siddht1** [comment]
what i suggest you guys is to rather connect it to a vpn or proxy to download the same,  whatever you do your actual ip broadcasted will be that of the ISP  which will not change.  You might even get a warning to use your bandwidth fairly.   Just report youtube that someone else on your network misused it and you are not involved .   selecting random ipv6 will also be a security issue and misuse of the entire network.

My suggestion wait as youtube will white list you soon (unless it thinks it to be a ddos attack) ,   and use any proxy or vpn for a while

**Comment by Bryannn** [comment]
What I was thinking was. Let's say I'm sending 10K requests a day. If it chose the best IP then that IP is sending 10K requests and will get blocked super quick. But if it chose a random IP, then 10K requests get divided by 256 that is 39 request per IP. Hence it might not get blocked. So that is why I think this random selection will be a good feature.

**Comment by yan12125** [comment]
I guess YouTube (and possible many other websites) counts requests by subnets rather than individual IPs, as changing IP is so easy. The hypothesis is easy to test: once an IP is blocked, change to another IP with ```--source-address```.

**Comment by Bryannn** [comment]
--source-address will work for ipv6?

**Comment by yan12125** [comment]
Yes. In fact ```--force-ipv6``` is a shortcut to ```--source-address ::```

**Comment by Bryannn** [comment]
Alright then. Thanks a lot for your help. Highly Appreciated!

**Comment by Bryannn** [comment]
Yan can we talk privately?

**Comment by yan12125** [comment]
You can drop me an email.

**Comment by Bryannn** [comment]
Hi Yan, I just sent you an email with this subject: Randomly select an available IP address for --source-address 

Please check.

Thanks!
